page-objects/pages/TASKListPage.js

```diff
@@ -159,60 +159,58 @@ async function clickDeleteTaskButtonInPopup() {
-	await waitHelpers.waitUntilNuxtClientLoads();
-	await driver.pause(5000);
-	let clickOnThatTask = (await getTaskFromNuxtClient(taskName)).toString();
-	await elementHelpers.scrollToElement(clickOnThatTask);
-	await elementHelpers.clickAndWait(clickOnThatTask);
+	await driver.pause(3000);
+	let clickOnThatTask = (await getTaskFromTaskOverview(taskName));
+	if (clickOnThatTask === undefined || clickOnThatTask != undefined){
+		if (clickOnThatTask.isDisplayedInViewport()) {
+			await elementHelpers.clickAndWait(clickOnThatTask);
+		}else{
+			await elementHelpers.scrollToElement(clickOnThatTask);
+			await elementHelpers.clickAndWait(clickOnThatTask);
+		}
+	}
 }
 
-async function getTaskFromNuxtClient(taskName) {
+async function getTaskFromTaskOverview(taskName) {
+	let tasksOnThePage = [];
 	await driver.pause(3000);
 	await waitHelpers.waitUntilElementIsVisible(taskOverviewLoad);
-	await driver.pause(3000);
-	const taskOverviewResult = await getNuxtTaskList();
-	const taskIndex = taskOverviewResult.indexOf(taskName);
-	let clickOnTask = taskOverviewResult[taskIndex];
-	const taskInTheList = '//div[text() =' + "'" + clickOnTask + "'" + ']';
-	return taskInTheList;
+	await driver.$(taskSection).$$(taskTitle).forEach(async function (element)  {
+		tasksOnThePage.push(await element.getText());
+	})
+	let isTaskInTheList = new Boolean(false);
+	isTaskInTheList =  (tasksOnThePage.includes(taskName)) ? driver.$(mod_extsprintf.sprintf(taskTitleText, taskName)) : false ;
+	return isTaskInTheList
 }
 
 async function taskDisplayed(taskName) {
-	let taskInTheList = (await getTaskFromNuxtClient(taskName)).toString();
+	let taskInTheList = (await getTaskFromTaskOverview(taskName));
 	await waitHelpers.waitUntilElementIsPresent(taskInTheList);
 }
 
 async function taskNotDisplayed(taskName) {
-	let taskInTheList = (await getTaskFromNuxtClient(taskName)).toString();
-	await waitHelpers.waitUntilElementIsNotPresent(taskInTheList);
+	let taskInTheList = (await getTaskFromTaskOverview(taskName));
+	expect(taskInTheList).to.equal(false);
 }
 
 async function studentSubmittedTask() {
 	await elementHelpers.clickAndWait(submittedTask);
 	await elementHelpers.clickAndWait(studentSubmitTask);
 }
 
-async function getNuxtTaskList() {
-	const listOfAllNuxtTasks = [];
-	let elements = await driver.$(taskSection);
-	await elements.$$(taskTitle).map(async function (element) {
-		listOfAllNuxtTasks.push(await element.getText());
-	});
-	return listOfAllNuxtTasks;
-}
-
 async function hoverOverTaskAndClickMenu(taskName) {
-	await driver.pause(5000);
+	await driver.pause(3000);
 	let taskTitle = await driver.$(mod_extsprintf.sprintf(taskTitleText, taskName));
-	await taskTitle.scrollIntoView(false);
-	if (await taskTitle.isDisplayedInViewport()){
-		let xOffset = await taskTitle.getLocation('x');
-		let yOffset = await taskTitle.getLocation('y');
-		taskTitle.moveTo(xOffset, yOffset);
-		await driver.pause(5000);
+	if (taskTitle === undefined || taskTitle != undefined){
```

Comment:
If it's not undefined it must be not undefined?
Here it's checking if it's true or false? Why both, that makes it unnecessary.